pwa/src/service-worker/recording-mode.js

```diff
@@ -77,19 +86,25 @@ export function shouldRequestBeRecorded({ url, event }) {
 }
 
 /** Request handler during recording mode */
-export function handleRecordedRequest({ request, event }) {
-    const recordingState = self.clientRecordingStates[event.clientId]
+export class RecordingMode extends Strategy {
+    _handle(request, handler) {
+        const { event } = handler
+        const recordingState = self.clientRecordingStates[event.clientId]
```

Comment:
This handler is refactored to a Workbox route handling strategy to take advantage of the 'plugin' functionality